### PR TITLE
persist atoms automatically; auth screen layout

### PR DESCRIPTION
**Pull request by nick-danylchenko**

### CODE REVIEW

File: recoil/index.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+export const RecoilRootWithPersisance = ({ children }: { children: ReactNode }) => {
+	type RecoilStateInitializer = (s: MutableSnapshot) => void
+
+	const [init, setInit] = useState<RecoilStateInitializer | null>(null);
+
+	useEffect(() => {
+		restoreState().then(init => setInit(() => init));
+	}, []);
+
+	return init === null ? (
```
Comment (andriy-moskalov) [inline recoil/index.tsx:60]: you can integrate it with https://docs.expo.dev/versions/latest/sdk/app-loading/ that can tell splashScreen to be visible while other async staff is going on.

File: recoil/index.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			}
+			const value = JSON.parse(json);
+			snapshot.set(atom, value);
+		})
+	}
+}
+
+
+const PersistContainer = ({ children }: { children: ReactNode }) => {
+	useRecoilTransactionObserver_UNSTABLE(({ snapshot }) => persistState(snapshot));
```
Comment (andriy-moskalov) [inline recoil/index.tsx:45]: sounds like dangerous (UNSTABLE)

File: App.tsx [hunk 3]
```diff
@@ -180,3 +147,28 @@ const MainContainer = () => {
 };
 
 export default App;
+
+
+const Hub =  ({ navigation }: NativeStackScreenProps<RootStackParamList, "Hub">) => {
```
Comment (andriy-moskalov) [inline App.tsx:152]: move all additional components from app, it should be as thin as possible

File: recoil/index.tsx (hunk 2 above)
Comment (nick-danylchenko) [reply recoil/index.tsx:45]: it works

File: recoil/index.tsx (hunk 1 above)
Comment (nick-danylchenko) [reply recoil/index.tsx:60]: will try it later

File: recoil/index.tsx (hunk 2 above)
Comment (nick-danylchenko) [reply recoil/index.tsx:45]: thou this whole approach - rewriting everything in store on any atom change - is less than ideal, i'll try to rewrite it eventually

**Pull request merged**